### PR TITLE
docs: create api reference landing page

### PR DESCRIPTION
**Pull request by JohannesMessner**
**ToDo**:

- [x] Decide what should and should not be on this page
- [x] Add descriptive docstrings

### CODE REVIEW

**Comment by samsja** [comment]
I think that the list exposed object is currently nice. maybe @davidbp can think about something that we forgot.

File: docs/api.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+   :nosignatures:
+   :template: class.rst
+
+   document.Document
+   document.DocumentArray
+   array.chunk.ChunkArray
+   array.match.MatchArray
+
+
+:mod:`docarray.array` - Storage backends
```
Comment (hanxiao) [inline docs/api.rst:25]:
```suggestion
:mod:`docarray.array` - Document stores
```

**Comment by JohannesMessner** [comment]
Docstrings are not yet amazing, but ready for a first release I would say.

File: docarray/array/qdrant.py [hunk 2]
```diff
@@ -5,8 +5,40 @@
 
 
 class DocumentArrayQdrant(StorageMixins, DocumentArray):
-    """This is a :class:`DocumentArray` that uses Qdrant as
-    vector search engine and storage.
+    """
+    DocumentArray that stores Documents in a `Qdrant <https://weaviate.io/>`_ vector search engine.
+
+    .. note::
+        This feature requires `qdrant-client`. You can install it via `pip install "docarray[qdrant]"`.
```
Comment (samsja) [inline docarray/array/qdrant.py:12]: I would not say it is a feature maybe class or object ?


```suggestion
        This DocumentArray requires `qdrant-client`. You can install it via `pip install "docarray[qdrant]"`.
```

File: docarray/array/sqlite.py [hunk 3]
```diff
@@ -6,5 +6,32 @@
 
 
 class DocumentArraySqlite(StorageMixins, DocumentArray):
+    """
+    DocumentArray that stores Documents in a `SQLite database <https://www.sqlite.org/index.html>`_.
```
Comment (samsja) [inline docarray/array/sqlite.py:10]: I would add a line to say that the Document are store on disk and not in memory and that it is the simpler way to deal with persistent da

File: docarray/array/weaviate.py [hunk 4]
```diff
@@ -5,8 +5,34 @@
 
 
 class DocumentArrayWeaviate(StorageMixins, DocumentArray):
-    """This is a :class:`DocumentArray` that uses Weaviate as
-    vector search engine and storage.
+    """
+    DocumentArray that stores Documents in a `Weaviate <https://weaviate.io/>`_ vector search engine.
+
+    .. note::
+        This feature requires `weaviate-client`. You can install it via `pip install "docarray[weaviate]"`.
```
Comment (samsja) [inline docarray/array/weaviate.py:12]: same as before
```suggestion
        This DocumentArray requires `weaviate-client`. You can install it via `pip install "docarray[weaviate]"`.
```

**Review by JoanFM — APPROVED** [review]

File: docarray/array/annlite.py [hunk 5]
```diff
@@ -5,5 +5,45 @@
 
 
 class DocumentArrayAnnlite(StorageMixins, DocumentArray):
+    """
+    DocumentArray that stores Documents in `ANNLite <https://github.com/jina-ai/annlite>`_.
+
+    .. note::
+        This feature requires `annlite`. You can install it via `pip install "docarray[annlite]"`.
+
```
Comment (samsja) [inline docarray/array/annlite.py:13]:
```suggestion
        This DocumentArray requires `annlite`. You can install it via `pip install "docarray[annlite]"`.

```

Comment (samsja) [reply docarray/array/annlite.py:13]: I forgot this one in my review

**Review by samsja — APPROVED** [review]

**Review by samsja — APPROVED** [review]

**Pull request merged**